### ISSUE TITLE
Scheduler.py Error and Empty Output

### ISSUE THREAD

**Issue by JoshMumford** [UW-Hydro/MetSim#167]
I keep getting this error string whenever I try to run MetSim with my files. I'm hoping someone might be able to help me figure out what to change to correct this. There is a file output to the output directory with the correct dimensions and variables, but every index is NaN. Do I need to change something in my configuration file? Any help is greatly appreciated! Thanks in advance! (Here is a [link ](https://drive.google.com/file/d/1ZVwJ6JbhpXbN0clo_gSDduuIW7wQpBuf/view?usp=sharing)to download the input/conf files associated with this run)
![MetSimError_2019-07-11](https://user-images.githubusercontent.com/52769976/61078807-5a033800-a3d6-11e9-982f-2cfc6f2eb2f6.PNG)

**Comment by arbennett** [comment]
Hi @JoshMumford,

I am not sure what would cause this specific error, but it appears to be more of an error in the underlying library which handles parallelism. @jhamman - have you ever seen this error from `distributed` and have any insight on what might cause it?

However, I just tried running with your dataset and found a different issue than the one you have here. 
Running MetSim 2.0.1 from a fresh install using `conda install -c conda-forge metsim` gave me this error after about 3 minutes of runtime:

```
Exception: ValueError('Daily maximum temperature lower than daily minimum temperature!')  
```

Basically, the Livneh dataset didn't enforce that `T_min < T_max`, so it requires some preprocessing to use for some locations. I think it's possible that the same underlying problem is leading to us both having errors though. Can I ask what version of MetSim you are using and how you installed? You can check the version with `python -c "import metsim; print(metsim.__version__)"`.

**Comment by JoshMumford** [comment]
Hi @arbennett, I have v2.0.1 installed. I installed it using conda install -c conda-forge metsim.

**Comment by arbennett** [comment]
Okay, a couple more questions. Could you attach the output of the command `conda list`? Also, what kind of machine are you running MetSim on? Is it a shared machine or one that only you are using?

**Comment by JoshMumford** [comment]
Here is the output of conda list:
[conda_list.docx](https://github.com/UW-Hydro/MetSim/files/3398427/conda_list.docx)

The machine is a shared machine.

**Comment by arbennett** [comment]
It appears that you have version 1.26.0 of `distributed` and I am using version 2.1.0. Could you try upgrading `distributed` with `conda upgrade -c conda-forge distributed`?

**Comment by JoshMumford** [comment]
`conda upgrade -c conda-forge distributed` isn't upgrading distributed to v2.1.0. `conda list` still shows that v1.26.0 is installed after running without any errors thrown. Is there another way to update the distributed package I could try?

**Comment by arbennett** [comment]
It might be easier for you to just create a new conda environment for MetSim just by doing:
```
conda create -n metsim
conda activate metsim
conda install -c conda-forge metsim
```

**Comment by JoshMumford** [comment]
I apologize for the delay in communication. Now I am getting a different error after doing that (I've attached the new `conda list` output as well)
[condaList.docx](https://github.com/UW-Hydro/MetSim/files/3412835/condaList.docx)
![MetSimError_2019-07-19](https://user-images.githubusercontent.com/52769976/61567822-c2d15c80-aa35-11e9-8ed7-cddca20decc9.PNG)

**Comment by arbennett** [comment]
I wonder if maybe running a dask client manually will help. You can do this by running in a new python sesison:
```
from distributed import Client
client = Client()
client.write_scheduler_file('~/scheduler')
```

And then when you run MetSim use `ms LivMS_Test.conf -s ~/scheduler`.

**Comment by jhamman** [comment]
@JoshMumford - let's actually simplify everything even further. 

You can run MetSim in a single-threaded mode via this command:

```bash
ms LivMS_Test.conf -s single-threaded
```

This will obviously be much slower but is a great way to uncover bugs/errors that aren't easy to diagnose in a distributed parallel environment.

Provided that doesn't uncover anything, I suggest you run through the quickstart docs for dask-distributed: https://distributed.dask.org/en/latest/quickstart.html. What we want to see is whether you can run a simple dask-distributed application before we jump into the combined dask-metsim space.
